packages/fluentui/react-northstar/src/components/Attachment/Attachment.tsx

```diff
@@ -61,104 +70,161 @@ export interface AttachmentProps extends UIComponentProps, ChildrenComponentProp
+ * An Attachment represents a file or media attachment, which may contain some metadata or actions.
+ */
+const Attachment = compose<'div', AttachmentProps, AttachmentStylesProps, {}, {}>(
+  (props, ref, composeOptions) => {
+    const context: ProviderContextPrepared = React.useContext(ThemeContext);
+    const { setStart, setEnd } = useTelemetry(composeOptions.displayName, context.telemetry);
+    setStart();
+
+    const {
+      accessibility,
+      action,
+      actionable,
+      body,
       className,
+      description,
       design,
+      disabled,
+      header,
+      icon,
+      onClick,
+      progress,
       styles,
       variables,
-    }),
-    rtl: context.rtl,
-  });
-
-  const ElementType = getElementType(props);
-  const unhandledProps = useUnhandledProps(Attachment.handledProps, props);
-
-  const handleClick = (e: React.KeyboardEvent | React.MouseEvent) => {
-    if (disabled) {
-      e.preventDefault();
-      return;
-    }
-
-    _.invoke(props, 'onClick', e, props);
-  };
-
-  const element = (
-    <ElementType {...getA11Props('root', { className: classes.root, onClick: handleClick, ...unhandledProps })}>
-      {AttachmentIcon.create(icon)}
-
-      {(header || description) &&
-        AttachmentBody.create(body, {
-          overrideProps: {
-            content: (
-              <>
-                {AttachmentHeader.create(header)}
-                {AttachmentDescription.create(description)}
-              </>
-            ),
-          },
+    } = props;
+
+    const getA11Props = useAccessibility(accessibility, {
+      debugName: composeOptions.displayName,
+      actionHandlers: {
+        performClick: e => {
+          if (e.currentTarget === e.target) {
+            e.stopPropagation();
+            handleClick(e);
+          }
+        },
+      },
+      rtl: context.rtl,
+    });
+
+    const { classes } = useStyles<AttachmentStylesProps>(composeOptions.displayName, {
+      className: attachmentClassName,
+      mapPropsToStyles: () => ({
+        actionable: actionable || !!onClick,
+        disabled,
+      }),
+      mapPropsToInlineStyles: () => ({
+        className,
+        design,
+        styles,
+        variables,
+      }),
+      rtl: context.rtl,
+      composeOptions,
+      unstable_props: props,
+    });
+
+    const slotProps = composeOptions.resolveSlotProps<AttachmentProps>(props);
+
+    const ElementType = getElementType(props);
+    const unhandledProps = useUnhandledProps(composeOptions.handledProps, props);
+
+    const handleClick = (e: React.KeyboardEvent | React.MouseEvent) => {
+      if (disabled) {
+        e.preventDefault();
+        return;
+      }
+
+      _.invoke(props, 'onClick', e, props);
+    };
+
+    const element = (
+      <ElementType {...getA11Props('root', { className: classes.root, onClick: handleClick, ref, ...unhandledProps })}>
+        {createShorthand(composeOptions.slots.icon, icon, {
+          defaultProps: () => slotProps.icon || {},
```

Comment:
```suggestion
          defaultProps: () => slotProps.icon,
```

?

Reply:
This looks extremely weird

Reply:
Well we wanted this not to he function, so now we have to do this. If nothing is defined here it is undefined, or I guess we can go ahead and change the `createShorthand` factory again.. Maybe we should change it, what do you think?

Reply:
Fixed, with change in the factories